src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPClientExecutor.cs

```diff
@@ -335,24 +340,41 @@ private void SetupDatabaseParameters()
                 numThreads = 1;
             }
 
-            int numRecords = (int)Math.Pow(10, recordCountExponent);
+            // safeguard against arm64 table setup
+
+            if (this.CpuArchitecture == Architecture.Arm64 || string.IsNullOrEmpty(this.NumTables))
+            {
+                this.NumTables = "10";
+            }
 
-            string threadCount = numThreads.ToString();
-            string recordCount = numRecords.ToString();
+            int numRecords = (int)Math.Pow(10, recordCountExponent);
 
-            // update the threads & record count only if they are not defined
+            // update the threads, table count & record count only if they are not defined
             // recommended for balanced & in memory scenarios to utilize
             // programmatic setup
 
-            this.Threads = string.IsNullOrEmpty(this.Threads) ? threadCount : this.Threads;
-            this.RecordCount = string.IsNullOrEmpty(this.RecordCount) ? recordCount : this.RecordCount;
+            string durationSecs = TimeSpan.Parse(this.Duration).TotalSeconds.ToString();
+            string threads = string.IsNullOrEmpty(this.Threads) ? numThreads.ToString() : this.Threads;
+            this.RecordCount = string.IsNullOrEmpty(this.RecordCount) ? numRecords.ToString() : this.RecordCount;
+
+            // set arguments & path up based on prepare arguments in profile
+
+            this.sysbenchPrepareArguments = $@"oltp_common --tables={this.NumTables} --table-size={this.RecordCount} --mysql-db={this.DatabaseName} --mysql-host={this.ServerIpAddress} prepare";
+            this.sysbenchLoggingArguments = $"{this.Workload} --threads={threads} --tables={this.NumTables} --table-size={this.RecordCount} --mysql-db={this.DatabaseName} ";
+            this.sysbenchExecutionArguments = this.sysbenchLoggingArguments + $"--mysql-host={this.ServerIpAddress} --time={durationSecs} ";
+            this.sysbenchPath = this.PlatformSpecifics.Combine(this.sysbenchDirectory, SysbenchOLTPClientExecutor.SysbenchFileName);
         }
 
         private async Task PrepareMySQLDatabase(EventContext telemetryContext, CancellationToken cancellationToken)
         {
             SysbenchOLTPState state = await this.stateManager.GetStateAsync<SysbenchOLTPState>(nameof(SysbenchOLTPState), cancellationToken);
 
-            if (!state.DatabaseInitialized)
+            int curTables = state.NumTables;
+            int curRecords = state.RecordCount;
+            int numTables = int.Parse(this.NumTables);
+            int recordCount = int.Parse(this.RecordCount);
+
+            if (!this.SkipInitialize || numTables > curTables || recordCount > curRecords)
```

Comment:
Could we still do what we discussed? Checking if current table configuration is same? instead of adding another SkipInitialize?